### PR TITLE
Use `enable_language(HIP)`

### PR DESCRIPTION
**Pull request by aurianer**

### CODE REVIEW

File: cmake/pika_setup_hip.cmake [hunk 1]
```diff
@@ -12,6 +12,22 @@ if(PIKA_WITH_HIP AND NOT TARGET roc::rocblas)
...
   endif(PIKA_WITH_CUDA)
 
+  # Check HIP standard
+  if(DEFINED CMAKE_HIP_STANDARD AND NOT CMAKE_HIP_STANDARD STREQUAL
+                                     PIKA_WITH_CXX_STANDARD
+  )
+    pika_error(
+      "You've set CMAKE_HIP_STANDARD to ${CMAKE_HIP_STANDARD} and PIKA_WITH_CXX_STANDARD to ${PIKA_WITH_CXX_STANDARD}. Please unset CMAKE_HIP_STANDARD."
+    )
+  endif()
```
Comment (msimberg) [inline cmake/pika_setup_hip.cmake:22]: This should be updated to reflect the changes in #396.

Comment (aurianer) [reply cmake/pika_setup_hip.cmake:22]: yes already in todo list thanks!

File: cmake/pika_add_executable.cmake [hunk 2]
```diff
@@ -209,7 +209,11 @@ function(pika_add_executable name)
   endif()
 
   if(${name}_CUDA)
```
Comment (msimberg) [inline cmake/pika_add_executable.cmake:211]: We can probably change the option to `GPU` instead?
```suggestion
  if(${name}_GPU)
```

File: cmake/pika_setup_hip.cmake [hunk 3]
```diff
@@ -12,6 +12,23 @@ if(PIKA_WITH_HIP AND NOT TARGET roc::rocblas)
...
+    if(DEFINED CMAKE_HIP_STANDARD AND NOT CMAKE_HIP_STANDARD STREQUAL
+                                      PIKA_WITH_CXX_STANDARD
+    )
+      pika_error(
+        "You've set CMAKE_HIP_STANDARD to ${CMAKE_HIP_STANDARD} and PIKA_WITH_CXX_STANDARD to ${PIKA_WITH_CXX_STANDARD}. Please unset CMAKE_HIP_STANDARD."
+      )
+    endif()
+  endif()
+
+  set(CMAKE_HIP_STANDARD ${PIKA_WITH_CXX_STANDARD})
```
Comment (msimberg) [inline cmake/pika_setup_hip.cmake:26]:
```suggestion
    set(CMAKE_HIP_STANDARD ${PIKA_WITH_CXX_STANDARD})
  endif()

```

**Review by msimberg — CHANGES_REQUESTED** [review]
I suppose the minimum CMake (and HIP?) version may need changing, but we can figure that out separately.

File: cmake/pika_setup_hip.cmake [hunk 4]
```diff
@@ -23,15 +23,19 @@ if(PIKA_WITH_HIP AND NOT TARGET roc::rocblas)
     endif()
   endif()
 
+  cmake_minimum_required(VERSION 3.23 FATAL_ERROR)
```
Comment (msimberg) [inline cmake/pika_setup_hip.cmake:26]: Let's make this 3.21 in the main CMakeLists.txt?

File: cmake/pika_setup_hip.cmake (hunk 3 above)
Comment (msimberg) [reply cmake/pika_setup_hip.cmake]: `CMAKE_HIP_STANDARD` is still being set outside the `if`.

**Comment by aurianer** [comment]
bors try

**Comment by aurianer** [comment]
bors try

**Review by msimberg — APPROVED** [review]
Looks very good, thanks!

**Comment by aurianer** [comment]
Needs a last push still

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try-

**Comment by msimberg** [comment]
bors try

File: .github/workflows/linux_hip.yml [hunk 5]
```diff
@@ -21,11 +21,14 @@ jobs:
...
     container: pikaorg/pika-ci-hip:2
-
     steps:
     - uses: actions/checkout@v2
     - name: Update apt repositories for ccache
       run: apt update
+    - name: Setup CMake
+      uses: jwlawson/actions-setup-cmake@v1.12
+      with:
+        cmake-version: '3.24.1'
```
Comment (msimberg) [inline .github/workflows/linux_hip.yml:31]:
```suggestion
        cmake-version: '3.21.0'
```

File: .github/workflows/linux_coverage.yml [hunk 6]
```diff
@@ -21,6 +21,10 @@ jobs:
     - uses: actions/checkout@v2
     - name: Update apt repositories for ccache
       run: apt update
+    - name: Setup CMake
+      uses: jwlawson/actions-setup-cmake@v1.12
+      with:
+        cmake-version: '3.24.1'
```
Comment (msimberg) [inline .github/workflows/linux_coverage.yml:27]: I don't think we need to add this here. The non-HIP CI image already has 3.22.1.

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
bors merge

**Comment by msimberg** [comment]
bors try

**Comment by aurianer** [comment]
bors try

**Comment by msimberg** [comment]
bors merge

**Pull request merged**